tst/EnergyPlus/unit/InternalHeatGains.unit.cc

```diff
@@ -829,3 +830,176 @@ TEST_F(EnergyPlusFixture, InternalHeatGains_CheckThermalComfortSchedules)
+        "    10,                      !- High Temperature {C}",
+        "    0.5,                     !- Fraction Radiant",
+        "    Baseboard Heat;          !- End - Use Subcategory",
+
+        "  ZoneContaminantSourceAndSink:CarbonDioxide,",
+        "    CO2people,               !- Name",
+        "    Zone 1,                  !- Zone Name",
+        "    0.0001125,               !- Design Generation Rate{ m3 / s }",
+        "    Schedule1;               !- Schedule Name",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    EXPECT_FALSE(has_err_output());
+
+    bool ErrorsFound(false);
+
+    DataGlobals::NumOfTimeStepInHour = 1;                 // must initialize this to get schedules initialized
+    DataGlobals::MinutesPerTimeStep = 60;                 // must initialize this to get schedules initialized
+    ScheduleManager::ProcessScheduleInput(outputFiles()); // read schedules
+    DataEnvironment::DayOfYear_Schedule = 1;
+    DataEnvironment::DayOfMonth = 1;
+    DataEnvironment::DayOfWeek = 1;
+    DataGlobals::HourOfDay = 1;
+    DataGlobals::TimeStep = 1;
+    ScheduleManager::UpdateScheduleValues();
+
+    HeatBalanceManager::GetZoneData(ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+    HeatBalanceManager::AllocateHeatBalArrays();
+
+    InternalHeatGains::GetInternalHeatGainsInput(outputFiles());
+
+    EXPECT_EQ(DataHeatBalance::TotPeople, 1);
+    EXPECT_EQ(DataHeatBalance::TotLights, 1);
+    EXPECT_EQ(DataHeatBalance::TotElecEquip, 1);
+    EXPECT_EQ(DataHeatBalance::TotGasEquip, 1);
+    EXPECT_EQ(DataHeatBalance::TotHWEquip, 1);
+    EXPECT_EQ(DataHeatBalance::TotStmEquip, 1);
+    EXPECT_EQ(DataHeatBalance::TotOthEquip, 1);
+    EXPECT_EQ(DataHeatBalance::TotBBHeat, 1);
+
+    EnergyPlus::createFacilityElectricPowerServiceObject(); // Needs to happen before InitInternalHeatGains
+
+    // First time should be all good, because ZnRpt values intialize to zero
+    InternalHeatGains::InitInternalHeatGains();
+
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).LtsPower, 100.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).ElecPower, 150.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).GasPower, 200.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).HWPower, 250.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).SteamPower, 300.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).BaseHeatPower, 1500.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).CO2Rate, 0.0001125);
+
+    // Second time should should give the same answers, because everything should reset before accumulating
+    InternalHeatGains::InitInternalHeatGains();
+
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).LtsPower, 100.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).ElecPower, 150.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).GasPower, 200.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).HWPower, 250.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).SteamPower, 300.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).BaseHeatPower, 1500.0);
+    EXPECT_EQ(DataHeatBalance::ZnRpt(1).CO2Rate, 0.0001125);
+}
```

Comment:
Why isn't this showing that little error with no line feed after the last bracket?

Reply:
Oh, because it's not showing line 1006.